hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -803,7 +803,15 @@ def __rpow__(self, other):
 
 class NDArrayExpression(Expression):
 
+    @typecheck_method(item=oneof(int, tuple))
     def __getitem__(self, item):
+        ndim = self._ir.typ.ndim
```

Comment:
I think you should probably also be able to use self._type here? I'm not sure there's much of a difference except in potentially having to compute the IR type when you don't strictly need it.

Reply:
Good point, can def use that